### PR TITLE
Add plugin-credentials as plugin-encryption dependency

### PR DESCRIPTION
**Pull request by ianwremmel**

### CODE REVIEW

File: packages/plugin-credentials/src/credentials/common.js [hunk 1]
```diff
@@ -357,16 +377,16 @@ export default {
    */
   set(attrs, options) {
     if (isObject(attrs)) {
+      if (attrs.authorization) {
+        attrs = attrs.authorization;
+      }
+
       if (attrs.access_token) {
```
Comment (mwegman) [inline packages/plugin-credentials/src/credentials/common.js]: I don't know what 'authorization' looks like but will it contain 'access_token' like 'attrs' possible did in the past?

File: packages/plugin-user/src/user.js [hunk 2]
```diff
@@ -27,6 +28,29 @@ const User = SparkPlugin.extend({
   },
 
   /**
+   * Finishes the signup process for a user that signed up on a mobile device
+   * @param {Object} options
+   * @param {string} options.encryptedQueryString
+   * @returns {Promise<Object>}
+   */
+  activate(options) {
```
Comment (mwegman) [inline packages/plugin-user/src/user.js]: can't you just do: activiate(options = {}) {

Comment (ianwremmel) [inline packages/plugin-user/src/user.js]: default params isn't in es2015. Other than the explicit inclusion of decorators, the sdk has a fairly conservative babel config.

File: packages/plugin-credentials/src/credentials/common.js (hunk 1 above)
Comment (ianwremmel) [inline packages/plugin-credentials/src/credentials/common.js]: authorization, if a result of the spark-core credentials plugin will have an access_token. if it's a result of the legacy sdk credentials layer, it'll have a supertoken. Basically, these if statements cover all the shapes i can think of.

(I've told a few people this lately: getting the credentials api right is one of the major blockers for releasing 1.0.0)

**Pull request merged**